src/connector/src/source/mqtt/enumerator/mod.rs

```diff
@@ -48,44 +43,20 @@ impl SplitEnumerator for MqttSplitEnumerator {
         context: SourceEnumeratorContextRef,
     ) -> ConnectorResult<MqttSplitEnumerator> {
         let (client, mut eventloop) = properties.common.build_client(context.info.source_id, 0)?;
-
         let topic = properties.topic.clone();
-        let mut topics = HashSet::new();
-        if !topic.contains('#') && !topic.contains('+') {
-            topics.insert(topic.clone());
-        }
-
-        client
-            .subscribe(topic.clone(), rumqttc::v5::mqttbytes::QoS::AtMostOnce)
-            .await?;
-
-        let topics = Arc::new(RwLock::new(topics));
 
         let connected = Arc::new(AtomicBool::new(false));
         let connected_clone = connected.clone();
 
         let stopped = Arc::new(AtomicBool::new(false));
         let stopped_clone = stopped.clone();
 
-        let topics_clone = topics.clone();
         tokio::spawn(async move {
             while !stopped_clone.load(std::sync::atomic::Ordering::Relaxed) {
                 match eventloop.poll().await {
-                    Ok(Event::Outgoing(Outgoing::Subscribe(_))) => {
+                    Ok(Event::Incoming(Incoming::ConnAck(_))) => {
```

Comment:
I'm just testing the connection. I could test the subscribe, but then I would also get the messages. Also the current implementation tests the subscribe going out, not the subscribe acknowledge coming in, so this is comparable